### ISSUE TITLE
Gateway fails to read certificate

### ISSUE THREAD

**Issue by cfutro-bg** [TykTechnologies/tyk#3292]
**Branch/Environment/Version**
- Branch/Version: Stable 3.0
- Environment: All

**Describe the bug**
Gateway fails to read a certificate that was functional in tyk gateway 2.9.4

**Reproduction steps**

1. Add a .pem containing cert and private key to dashboard
2. Add the cert id to tyk.conf at http_server_options.ssl_certificates and set use_ssl true

**Actual behavior**
Gateway fails with the error:
`TLS handshake error from xxx.xxx.xxx.xxx:xxxxx: tls: no certificates configured`

Gateway additionally reports:
`level=warning msg="Reconnecting storage: Redis is either down or ws not configured" prefix=pub-sub`
But the certificate is visible in redis.

This behavior is observed both on a fresh installation of 3.0 connecting to a fresh docdb and redis cluster as well as upgrading and keeping the pre-existing data.

**Expected behavior**
Gateway starts correctly and receives ssl traffic from dashboard

**Comment by l3ender** [comment]
We ran into the same issue when updating to 3.0.0-1. We were able to workaround by updating our tyk.conf to use certificate files on the server as opposed to referencing a certificate ID from the dashboard.

```
"http_server_options": {
  "use_ssl": true,
  "certificates": [
    {
      "domain_name": "*.yoursite.com",
      "cert_file": "./new.cert.cert",
      "key_file": "./new.cert.key"
    }
  ]
}
```

(be sure to disable the `TYK_GW_HTTPSERVEROPTIONS_SSLCERTIFICATES` environment variable if you are using it).

**Comment by sedkis** [comment]
Confirmed bug.  Thanks for the workaround @l3ender

**Comment by andyo-tyk** [comment]
This was fixed in Tyk 3.0.9, Tyk 4.0.1 and Tyk 4.1.0.
